kidviz/views.py

```diff
@@ -69,7 +77,17 @@ def form_valid(self, form):
         self.request.session['constructs'] = [c.id for c in constructs]
         self.request.session['context_tags'] = [t.id for t in tags]
         self.request.session['curricular_focus'] = d['curricular_focus']
-        self.request.session['create_new'] = True
+
+        if 're_setup' in self.request.session:
+            remove_key_from_session(self.request, 're_setup')
+            draft_observation = Observation.objects.filter(is_draft=True).order_by('-id').first()
+
+            # Create new observation when course has change.
+            if draft_observation.course.id != course.id:
+                self.request.session['create_new'] = True
```

Comment:
Shouldn't we add special case for resetup? I think resetup should be a special case, not normal flow with creating observation template.
I mean, adding special case shouldnt involve many changes in default behaviour.